keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -260,6 +260,23 @@ Example Validation Rules:
 
 - Each validation rule has an optional 'message' field for the error message that
 will be surfaced when the validation rule evaluates to false.
+- As an alternative to the `message` field, there is also a
+`messageExpression` field that represents a CEL expression that evaluates to a
+string that will be used when the validation rule evaluates to false. If the
+`messageExpression` field does not evaluate to a string, then the CRD write will fail. If the
+runtime cost limit is exceeded during `messageExpression` execution, then this
+is logged. Whether or not the action is admitted after that depends upon failure policy.
+Additionally, as part of [the KEP update to add expression composition](https://github.com/kubernetes/enhancements/pull/3669/files),
+expressions defined under `variables` will be accessible from `messageExpression`.
+- There are several compile-time checks performed on CEL's `string.format` function that can affect `messageExpression`:
+  - If the formatting string passed to `string.format` is a constant, then the formatting string will be checked at
+  compile-time that it parses correctly.
+  - If the `arg` array passed to `string.format` is a literal, and the formatting string is a constant, then
+  at compile-time `arg`'s cardinality will be checked against the formatting string. Passing too few/too many arguments
```

Comment:
Just a reminder that any "compile-time" checking done on `string.format` args will need to be merged in beta. So we would need to merge it in 1.27 so that we can graduate to GA in 1.28.